test/summary.c

```diff
@@ -0,0 +1,214 @@
+#include <pg_query.h>
+#include "src/pg_query_summary.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// ARBITRARY_LENGTH_LIMIT is purely arbitrary to avoid unbounded strcmp() usage.
+// If it causes a problem, make the number bigger.
+// (Using strcmp() means writing a test wrong can cause infinite loops,
+//  which is very annoying to debug.)
+// -@duckinator
+const size_t ARBITRARY_LENGTH_LIMIT = 500;
+
+typedef struct {
+	size_t passed;
+	size_t failed;
+	size_t skipped;
+	MemoryContext last_ctx;
+} TestSuiteState;
+
+static TestSuiteState test_state = {0};
+
+typedef void (TestFn)(void);
+
+#define TEST_INIT() printf("%s (line %i)\n", __func__, __LINE__)
```

Comment:
As discussed, if we can make it work to have this in a helper file (`.c` or `.h`) so its reusable, that'd be great.